### ISSUE TITLE
Searching for parts of words is not working with the LuceneAdapter

### ISSUE THREAD

**Issue by ampaze** [sulu/sulu#4189]
| Q | A
| --- | ---
| Bug? | yes
| New Feature? | no
| Sulu Version | 1.6.21

#### Actual Behavior

Searching for "faktor" does not give any results, while "faktoren" does.

Sulu builds a search query that Lucene does not understand correctly. 

What sulu builds:
> "+("faktor" OR "faktor*" OR "faktor~") "

results in Lucene looking for exact match "faktor" three times, as  ZendSearch\Lucene\Search\Query\Preprocessing\Phrase is stripping * and ~

#### Expected Behavior

Sulu should generate a search query that Lucene understands.

#### Steps to Reproduce

Search for a part of a word that is in the index.

**Comment by danrot** [comment]
Hm, that's strange... Just tested it, and I can confirm that it doesn't work. But I also tested the search in the Admin, and there it works when I enter `Faktor*`, so it doesn't seem to strip the asterisk there...

If I remember correctly, we put quite some time into this, so that it is working with ZendSearch and ElasticSearch... If you come up with another query which works in both situations we are happy to merge it :-)

**Comment by ymc-thzi** [comment]
@danrot having kind of the same issue. In the admin I can search for "port*" and I get the results for pages with "portal" inside the content. If I do the same on the website I get 0 results.
(I rebuilt, reindexed with each console on each env)
What is the difference? I would love to have the same searchresults on the website...

**Comment by danrot** [comment]
In general we have said that in the admin there are more technical affin users, and there is the description that you can use an asterisk as a placeholder. However, we don't demand the same amount of knowledge from a website visitor, so we decided to adapt the query in a way, that allows to search for parts of words more easily (I think that's also what people are used to from other search forms).

But as said before, if you are searching for just "port" on the website, you should receive the pages with "portal" as well...

**Comment by ymc-thzi** [comment]
> if you are searching for just "port" on the website, you should receive the pages with "portal" as well...
@danrot Thanks for the reply, but that is not the case. If I search for just "port" I get 0 results. 
I solved it writing my own Symfony service querying massive search directly.

**Comment by danrot** [comment]
@ymc-thzi I said it should, not that it does :see_no_evil: And how do you query it? Does that also work for elastic search?

**Comment by AlexandraDeMatos** [comment]
@ymc-thzi @danrot Hello ! I have the same problem. Can you tell me in details how to do your own Symfony service querying massive search ? I search an issu since a week. Thanks by advance.

**Comment by ymc-thzi** [comment]
Hi @AlexandraDeMatos 
I am using elastic search. And I setup a common Symfony service class that is called from a common Symfony controller that is processing the post param from a custom twig template search form. I have even overwritten the predefined sulu route named "search" like so:
**/app/config/website/routing.yml**
```
someWebsiteName:
    resource: '../../../src/AppBundle/Controller/'
    type:     annotation

sulu_media:
    resource: "@SuluMediaBundle/Resources/config/routing_website.yml"
...
```
**/src/AppBundle/Controller/SearchController.php**
```
class SearchController extends Controller

{

    /**
     * @Route("/search", name="search")
     *
     */
    public function searchAction(Request $request)
    {
/// ... call search service in here...
```
**src/AppBundle/Service/SearchService.php**
```
...
    /**
     * @var array
     * Allowed Sulu indexes: "snippet", "media", "page_someWebsiteName_published", "contact", "account", "page_someWebsiteName", "page"
     */
    protected $indexes = [
        'page',
        'page_someWebsiteName_published',
        //'media' //@TODO: enable this when filling search results with images and documents
    ];

...
    /**
     *
     * @return array
     */
    public function search($query)
    {

        $searchManager = $this->container->get('massive_search.search_manager');

        $hits = $searchManager
            ->createSearch(self::prepareQuery($query))
            ->indexes($this->indexes)
            ->setLimit($this->limit)
            ->execute();

        $results = $hits->getArrayCopy();
...
```
The static method `prepareQuery` is cleaning the string, like deleting quotes and add an asterisk after each word. It is still WIP so I don't post it here, but adding an asterisk manually after each word will work to find only parts of words. In the $results you get a whole lot of sulu entities stuff which can be given back to the template via the controller. I am even planning to enrich this result with images and documents by getting the sulu objects from the media or document id...

**Comment by SimonStasius** [comment]
I had the same issue with Lucene and ElasticSearch.

As mentioned above, Sulu builds the query_string: `"+("faktor" OR "faktor*" OR "faktor~") "`
Using this in the websiteconsole with massive:search:query returns also 0 results.

So I tried a query_string with less quotation marks: `"+(faktor OR faktor* OR faktor~) "`
This works for me on website and console (with Lucene and ElasticSearch).

**Comment by danrot** [comment]
@SimonStasius If that works for you, could you provide a PR which changes to that version? I would like to test it then with Zend Lucene and ElasticSearch.

**Comment by niklasnatter** [comment]
Fixed by #5407
